src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ReactNode, useState } from 'react';
+import Button from '@mui/material/Button';
+import ConfirmationDialog from './Confirmation/ConfirmationDialog';
+
+export type DialogProps = {
+  children: ReactNode;
+  onCancel: () => void;
+  onConfirm?: () => void;
+  sx?: { [key: string]: string | number };
+  onSubmit: () => void;
+};
+
+const Dialog: React.FC<DialogProps> = ({ onSubmit, onCancel, onConfirm, children, sx }) => {
+  const [isOpen, setIsOpen] = useState(false); // State to manage the open status of the confirmation dialog
+  
+  const handleOpen = () => setIsOpen(true);
+  const handleClose = () => setIsOpen(false);
+  
+  const handleConfirm = () => {
+    if (onConfirm) {
+      onConfirm();
+    }
+    handleClose(); // Close the confirmation dialog after confirming
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (onConfirm) {
+      handleOpen(); // Open the confirmation dialog
+    } else {
+      onSubmit(); // If no confirmation needed, proceed with submit
+    }
+  };
+
+  return (
+    <div style={sx}>

```

Comment:
I recommend using spread here to ensure you can set baseline styles that can be overridden:
```
style={{
  position: 'absolute',
  top: '15%',
  right: '15%',
  bottom: '15%',
  left: '15%',
  ...sx
}}
```